From b91b460b943185a1585cbbfe331eac38527b64cb Mon Sep 17 00:00:00 2001
From: Steve Grubb <ausearch.1@gmail.com>
Date: Wed, 28 Jan 2026 22:07:10 -0500
Subject: [PATCH] whitespace fix
Content-type: text/plain

---
 src/library/rpm-backend.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/library/rpm-backend.c b/src/library/rpm-backend.c
index 5b24c9589040..c4a90cd1c9f1 100644
--- a/src/library/rpm-backend.c
+++ b/src/library/rpm-backend.c
@@ -443,7 +443,7 @@ out:
 	// cleaning up
 	struct _hash_record *item, *tmp;
 	HASH_ITER( hh, hashtable, item, tmp) {
-		   HASH_DEL( hashtable, item );
+		HASH_DEL( hashtable, item );
 		free((void*)item->key);
 		free((void*)item);
 	}
-- 
2.53.0

