From 67c116d07ed4e73127392a2100a042882488585a Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Tue, 27 Sep 2022 10:32:28 -0400
Subject: [PATCH] Detect trusted static apps running programs by ld.so

---
 ChangeLog           | 1 +
 src/library/event.c | 1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/library/event.c b/src/library/event.c
index cbb4292..4d79eb9 100644
--- a/src/library/event.c
+++ b/src/library/event.c
@@ -149,7 +149,6 @@ int new_event(const struct fanotify_event_metadata *m, event_t *e)
 				skip_path = 1;
 			}
 			evict = 0;
-			skip_path = 1;
 			subject_reset(s, EXE);
 			subject_reset(s, COMM);
 			subject_reset(s, EXE_TYPE);
