diff -up ./src/cli/fapolicyd-cli.c.segfault ./src/cli/fapolicyd-cli.c
--- ./src/cli/fapolicyd-cli.c.segfault	2022-08-03 17:51:54.903081124 +0200
+++ ./src/cli/fapolicyd-cli.c	2022-08-03 17:55:18.256458750 +0200
@@ -77,6 +77,7 @@ static struct option long_opts[] =
 	{"ftype",	1, NULL, 't'},
 	{"list",	0, NULL, 'l'},
 	{"update",	0, NULL, 'u'},
+	{NULL,	0, NULL, 0 }
 };
 
 static const char *_pipe = "/run/fapolicyd/fapolicyd.fifo";
