From 686096c400587d5f28e041c4ebb58e07e4d7b3fc Mon Sep 17 00:00:00 2001
From: Radovan Sroka <rsroka@redhat.com>
Date: Wed, 20 Nov 2024 14:30:14 +0100
Subject: [PATCH] Fix creation of RUN_DIR because it breaks rpm verify

Signed-off-by: Radovan Sroka <rsroka@redhat.com>
---
 src/library/database.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/library/database.c b/src/library/database.c
index 44dd0b2b..d89b16c1 100644
--- a/src/library/database.c
+++ b/src/library/database.c
@@ -122,6 +122,19 @@ int preconstruct_fifo(const conf_t *config)
 		    strerror_r(errno, err_buff, BUFFER_SIZE));
 		return 1;
 	} else {
+
+		if ((chmod(RUN_DIR, 0770))) {
+			msg(LOG_ERR, "Failed to fix mode of dir %s (%s)",
+			    RUN_DIR, strerror_r(errno, err_buff, BUFFER_SIZE));
+			return 1;
+		}
+
+		if ((chown(RUN_DIR, 0, config->gid))) {
+			msg(LOG_ERR, "Failed to fix ownership of dir %s (%s)",
+			    RUN_DIR, strerror_r(errno, err_buff, BUFFER_SIZE));
+			return 1;
+		}
+
 		/* Make sure that there is no such file/fifo */
 		unlink_fifo();
 	}
