From 6ce38340bd0ed3863259224de613ef6a29ca9297 Mon Sep 17 00:00:00 2001
From: Radovan Sroka <rsroka@redhat.com>
Date: Sun, 20 Jul 2025 20:56:34 +0200
Subject: [PATCH] ddasd

Signed-off-by: Radovan Sroka <rsroka@redhat.com>
---
 src/library/fapolicyd-backend.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/library/fapolicyd-backend.h b/src/library/fapolicyd-backend.h
index 2844780..26f0a54 100644
--- a/src/library/fapolicyd-backend.h
+++ b/src/library/fapolicyd-backend.h
@@ -32,7 +32,8 @@
 typedef enum { SRC_UNKNOWN, SRC_RPM, SRC_FILE_DB, SRC_DEB } trust_src_t;
 
 // source, size, sha
-#define DATA_FORMAT "%u %lu %64s"
+// do not pad the hash value so SHA1 and SHA256 digests parse correctly
+#define DATA_FORMAT "%u %lu %s"
 
 typedef struct _backend
 {
-- 
2.50.1

