diff -up ./fapolicyd.spec.rpm-loader ./fapolicyd.spec
--- ./fapolicyd.spec.rpm-loader	2024-04-29 17:59:11.000000000 +0200
+++ ./fapolicyd.spec	2025-05-18 20:11:39.722717947 +0200
@@ -268,6 +268,7 @@ fi
 %attr(644,root,root) %{_tmpfilesdir}/%{name}.conf
 %attr(755,root,root) %{_sbindir}/%{name}
 %attr(755,root,root) %{_sbindir}/%{name}-cli
+%attr(755,root,root) %{_sbindir}/%{name}-rpm-loader
 %attr(755,root,root) %{_sbindir}/fagenrules
 %attr(644,root,root) %{_mandir}/man8/*
 %attr(644,root,root) %{_mandir}/man5/*
diff -up ./src/daemon/fapolicyd.c.rpm-loader ./src/daemon/fapolicyd.c
--- ./src/daemon/fapolicyd.c.rpm-loader	2025-05-18 20:11:39.721683468 +0200
+++ ./src/daemon/fapolicyd.c	2025-05-18 20:11:39.722858226 +0200
@@ -105,17 +105,19 @@ static void install_syscall_filter(void)
 	ctx = seccomp_init(SCMP_ACT_ALLOW);
 	if (ctx == NULL)
 		goto err_out;
-
+#ifndef USE_RPM
 	rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EACCES),
 				SCMP_SYS(execve), 0);
 	if (rc < 0)
 		goto err_out;
-#ifdef HAVE_FEXECVE
-# ifdef __NR_fexecve
+
+# ifdef HAVE_FEXECVE
+#  ifdef __NR_fexecve
 	rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EACCES),
 				SCMP_SYS(fexecve), 0);
 	if (rc < 0)
 		goto err_out;
+#  endif
 # endif
 #endif
 	rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EIO),
diff -up ./src/handler/fapolicyd-rpm-loader.c.rpm-loader ./src/handler/fapolicyd-rpm-loader.c
--- ./src/handler/fapolicyd-rpm-loader.c.rpm-loader	2025-05-18 20:11:39.722986504 +0200
+++ ./src/handler/fapolicyd-rpm-loader.c	2025-05-18 20:13:48.417481732 +0200
@@ -0,0 +1,85 @@
+/*
+ * fapolicy-rpm-loader.c - loader tool for fapolicyd
+ * Copyright (c) 2025-2025 Red Hat Inc.
+ * All Rights Reserved.
+ *
+ * This software may be freely redistributed and/or modified under the
+ * terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING. If not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335, USA.
+ *
+ * Authors:
+ *   Radovan Sroka <rsroka@redhat.com>
+ */
+
+#include "config.h"
+#include <stdio.h>
+#include <string.h>
+#include <sys/syslog.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+#include <ctype.h>
+#include <magic.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <stdatomic.h>
+#include <lmdb.h>
+#include <limits.h>
+#include <signal.h>
+
+#include "backend-manager.h"
+#include "daemon-config.h"
+#include "message.h"
+#include "llist.h"
+#include "fd-fgets.h"
+#include "paths.h"
+
+volatile atomic_bool stop = 0;  // Library needs this
+unsigned int debug_mode = 0;			// Library needs this
+unsigned int permissive = 0;			// Library needs this
+
+
+int do_rpm_init_backend(void);
+int do_rpm_load_list(conf_t * conf);
+int do_rpm_destroy_backend(void);
+
+extern backend rpm_backend;
+
+int main(int argc, char * const argv[])
+{
+
+	set_message_mode(MSG_STDERR, DBG_YES);
+
+	conf_t config;
+
+	load_daemon_config(&config);
+
+	do_rpm_init_backend();
+	do_rpm_load_list(&config);
+
+	msg(LOG_INFO, "Loaded files %ld", rpm_backend.list.count);
+
+	list_item_t *item = list_get_first(&rpm_backend.list);
+	for (; item != NULL; item = item->next) {
+		printf("%s %s\n", (const char*)item->index, (const char*)item->data);
+	}
+
+	do_rpm_destroy_backend();
+
+	free_daemon_config(&config);
+	return 0;
+}
+
diff -up ./src/library/rpm-backend.c.rpm-loader ./src/library/rpm-backend.c
--- ./src/library/rpm-backend.c.rpm-loader	2025-05-18 20:11:39.718862697 +0200
+++ ./src/library/rpm-backend.c	2025-05-18 20:11:39.723077584 +0200
@@ -23,8 +23,11 @@
  */
 
 #include "config.h"
+#include <ctype.h>
 #include <stddef.h>
 #include <sys/types.h>
+#include <sys/wait.h>
+#include <spawn.h>
 #include <rpm/rpmlib.h>
 #include <rpm/rpmts.h>
 #include <rpm/rpmmacro.h>
@@ -37,11 +40,16 @@
 
 #include "message.h"
 #include "gcc-attributes.h"
+#include "fd-fgets.h"
 #include "fapolicyd-backend.h"
 #include "llist.h"
 
 #include "filter.h"
 
+int do_rpm_init_backend(void);
+int do_rpm_load_list(const conf_t *);
+int do_rpm_destroy_backend(void);
+
 static int rpm_init_backend(void);
 static int rpm_load_list(const conf_t *);
 static int rpm_destroy_backend(void);
@@ -183,9 +191,104 @@ struct _hash_record {
 	UT_hash_handle hh;
 };
 
-extern unsigned int debug_mode;
+#define BUFFER_SIZE 4096
+#define MAX_DELIMS 3
 static int rpm_load_list(const conf_t *conf)
 {
+	int pipefd[2];
+	if (pipe(pipefd) == -1) {
+		perror("pipe failed");
+		return 1;
+    }
+	// we well read stdout later
+	// there will be data from rpmdb
+	posix_spawn_file_actions_t actions;
+	posix_spawn_file_actions_init(&actions);
+	posix_spawn_file_actions_addclose(&actions, pipefd[0]);
+	posix_spawn_file_actions_adddup2(&actions, pipefd[1], STDOUT_FILENO);
+	posix_spawn_file_actions_addclose(&actions, pipefd[1]);
+
+	char *argv[] = { NULL };
+	char *custom_env[] = { NULL };
+
+	pid_t pid = -1;
+	// int status = posix_spawn(&pid, "/home/rsroka/Work/fapolicyd-upstream-fork2/src/fapolicyd-rpm-loader", &actions, NULL, argv, custom_env);
+	int status = posix_spawn(&pid, "/usr/sbin/fapolicyd-rpm-loader", &actions, NULL, argv, custom_env);
+
+
+	close(pipefd[1]);  // Parent doesn't write
+
+	if (status == 0) {
+		msg(LOG_DEBUG, "fapolicyd-rpm-loader spawned with pid: %d", pid);
+
+		char buff[BUFFER_SIZE];
+		fd_fgets_context_t * fd_fgets_context = fd_fgets_init();
+		do {
+			fd_fgets_rewind(fd_fgets_context);
+			int res = fd_fgets(fd_fgets_context, buff, sizeof(buff), pipefd[0]);
+			if (res == -1)
+				break;
+			else if (res > 0) {
+				char* end  = strchr(buff, '\n');
+
+				if (end == NULL) {
+					msg(LOG_ERR, "Too long line?");
+					continue;
+				}
+
+				int size = end - buff;
+				*end = '\0';
+
+				// it's better to parse it from the end because there can be space in file name
+				int delims = 0;
+				char * delim = NULL;
+				for (int i = size-1 ; i >= 0 ; i--) {
+					if (isspace(buff[i])) {
+						delim = &buff[i];
+						delims++;
+					}
+
+					if (delims >= MAX_DELIMS) {
+						buff[i] = '\0';
+						break;
+					}
+				}
+
+				char * index = strdup(buff);
+				char * data = strdup(delim + 1);
+				if (!index || !data) {
+					free(index);
+					free(data);
+					continue;
+				}
+
+				list_append(&rpm_backend.list, index, data);
+			}
+		} while(!fd_fgets_eof(fd_fgets_context));
+
+		fd_fgets_destroy(fd_fgets_context);
+
+		close(pipefd[0]);
+		waitpid(pid, NULL, 0);
+		
+	} else {
+		msg(LOG_ERR, "posix_spawn failed: %s\n", strerror(status));
+	}
+
+	posix_spawn_file_actions_destroy(&actions);
+
+	if (rpm_backend.list.count == 0) {
+		msg(LOG_DEBUG, "Recieved 0 files from rpmdb loader");
+		return 1;
+	}
+
+	return 0;
+}
+
+// this function is used in fapolicyd-rpm-loader
+extern unsigned int debug_mode;
+int do_rpm_load_list(const conf_t *conf)
+{
 	int rc;
 	unsigned int msg_count = 0;
 
@@ -298,6 +401,14 @@ static int rpm_load_list(const conf_t *c
 
 static int rpm_init_backend(void)
 {
+	list_init(&rpm_backend.list);
+
+	return 0;
+}
+
+// this function is used in fapolicyd-rpm-loader
+int do_rpm_init_backend(void)
+{
 	if (filter_init())
 		return 1;
 
@@ -318,3 +429,10 @@ static int rpm_destroy_backend(void)
 	list_empty(&rpm_backend.list);
 	return 0;
 }
+
+// this function is used in fapolicyd-rpm-loader
+int do_rpm_destroy_backend(void)
+{
+	list_empty(&rpm_backend.list);
+	return 0;
+}
diff -up ./src/Makefile.am.rpm-loader ./src/Makefile.am
--- ./src/Makefile.am.rpm-loader	2025-05-18 20:11:39.718965116 +0200
+++ ./src/Makefile.am	2025-05-18 20:11:39.723185202 +0200
@@ -75,6 +75,15 @@ libfapolicyd_la_SOURCES += \
 	library/filter.c \
 	library/filter.h
 
+sbin_PROGRAMS += fapolicyd-rpm-loader
+
+fapolicyd_rpm_loader_SOURCES = \
+	handler/fapolicyd-rpm-loader.c
+
+fapolicyd_rpm_loader_CFLAGS = $(fapolicyd_CFLAGS)
+fapolicyd_rpm_loader_LDFLAGS = $(fapolicyd_LDFLAGS)
+
+fapolicyd_rpm_loader_LDADD = libfapolicyd.la
 endif
 
 if WITH_DEB
